include/uuid/UUIDv1.h

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2023 General Motors GTO LLC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#ifndef _UUIDV1_H_
+#define _UUIDV1_H_
+#include <uuid/uuid.h>
+#include <cassert>
+#include <iostream>
+#include <string>
+
+//#include "../tools/tools.h"
+#include "spdlog/spdlog.h"
+
+namespace uprotocol::uuid::factory {
+
+using UUIDv1 = uint8_t*;
+
+/**
+ * Universally Unique Identifier (UUID) is a 128-bit label used for information
+ * in computer systems. 
+ * UUID Version-1 is based on the current time and the MAC
+ * address for the computer or "node" generating the UUID.
+ *
+ * UUIDv1 is generated using standard libuuid library.
+ * More information on
+ * https://www.ietf.org/rfc/rfc4122.txt
+ *
+ */
+class UUIDv1Factory {
+ public:
+  UUIDv1Factory() { uuidv1_ = new u_int8_t[uuidSize_]; }
```

Comment:
why do you need to allocate the buffer? 
Also please do not allocate memory in the constructor - as the constructor does not have return values and you cannot check if the allocation was successful

Reply:
also if you are allocating the buffer - who is responsible to release it?